core/lib/spree/event/subscribers.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Spree
+  module Event
+    class Subscribers
+      attr_reader :mappings
```

Comment:
`mappings` seems like an implementation detail concept, I'd avoid using it as the main public interface of this class.

Reply:
I made a quick attempt [here](https://github.com/solidusio/solidus/commit/d786b56428eb5e55754523bb53aba60aa68f1a4d) to explain what I was thinking (and being sure I'm not missing something big). Let me know your thoughts!

Reply:
The downside of my proposal is that we are exposing all the hash methods to users that in theory will be able to play too much with this mapping of subscribers. But at the end of the day, it's the same effect we have making `mappings` public, which is a hash.

Reply:
I think subclassing core ruby classes is generally considered an antipattern. A couple of old articles on that topic: 
* https://tenderlovemaking.com/2014/06/02/yagni-methods-are-killing-me.html 
* https://steveklabnik.com/writing/beware-subclassing-ruby-core-classes